### ISSUE TITLE
Should Output throw an exception in getValue() if stage is too low?

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#603]
Could the underlying computation throw the exception? The check may be redundant.

**Comment by aseth1** [comment]
Yes the Output should throw. The exception thrown by the computation will be too low-level in my opinion. It should be clear that an output was requested that was not realized or could not be evaluated due to the stage of the system.

**Comment by chrisdembia** [comment]
That's a good point...the error message should be clear.

The issue I ran into was that I had incorrectly specified the output's dependsOn stage. It was incorrect, and I got that exception even though the calculation would have been fine.

Another option is for the Output::getValue() to catch `StageTooLow` exceptions and give a more useful message.

I'm still not convinced that the inputs and outputs need stages associated with them.

**Comment by sherm1** [comment]
> I'm still not convinced that the inputs and outputs need stages associated with them.

As long as the stages are checked at run time (i.e. values must be available when asked for) it's probably not essential to check earlier. For most Simbody methods the required stage is just part of the documentation.

**Comment by jenhicks** [comment]
@chrisdembia Is this still an issue?

**Comment by chrisdembia** [comment]
I think so, as long as we still discuss whether outputs/inputs should have stages associated with them at all.

**Comment by jenhicks** [comment]
@aymanhab Idea is an enum to keep track of staging to provide user info for troubleshooting.

**Comment by jenhicks** [comment]
Moving to next release unless it becomes more of an issue.
